packages/flame_devtools/lib/widgets/position_component_attributes_form.dart

```diff
@@ -20,6 +20,10 @@ class PositionComponentAttributesForm extends ConsumerWidget {
       ),
     );
 
+    // We don't want to keep the attributes in cache since the user will most
+    // likely change them often.
+    ref.invalidate(positionComponentAttributesProvider(componentId));
```

Comment:
Are there more places listening to this provider? Otherwise it will dispose itself once it's not listened to in the tree anymore. I think adding invalidate in the build method is probably not really how it's meant to be used.

Reply:
It is used only here. Do you have any suggestion on how to make this provider never cache? I couldn't find any alternative 🙃

Reply:
When the widget leaves the tree the state is disposed if it is not listened to, so it shouldn't be cached if it is only used here? :thinking:

Reply:
I thought that too. But for some reason, the provider function is being executed only once per component, so if the user changed any values, selected a different component, then returned to the one they changed before, they would see the initial values before their changes, because somehow the provider were cached

Reply:
Aaah, autodispose is missing from here:
https://github.com/flame-engine/flame/pull/3263/files#diff-d87cdf5db8e0f1fb3c312e7b2cd3425ca724554c5e01791cdd23033cc299dd42R5

Reply:
Right, changed to autoDispose and it is better now!